### ISSUE TITLE
[Linux compilation] fatal error: too many errors emitted, stopping now

### ISSUE THREAD

**Issue by KAMiKAZOW** [SourMesen/Mesen-S#8]
```
[  451s] mkdir -p Core/obj.x64 && cd Core/obj.x64 && clang++ -fPIC -Wall --std=c++14 -O3  -m64 -flto  -c  ../Ppu.cpp
[  466s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  466s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  466s]                        ^
[  466s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, true, true>' requested here
[  466s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  466s]                 ^
[  466s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, true>' requested here
[  466s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  466s]                 ^
[  466s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true>' requested here
[  466s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  466s]                 ^
[  466s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  466s]                         RenderMode7<true>();
[  466s]                         ^
[  466s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  466s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  466s]                               ^
[  466s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  466s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  466s]                        ^
[  466s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, true, false>' requested here
[  466s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  466s]                 ^
[  466s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, true>' requested here
[  466s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  466s]                 ^
[  466s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true>' requested here
[  466s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  466s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, true>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, true, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types be[  429.793297] serial8250: too much work for irq4
[  467s] fore C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', true, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, true, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:443:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:443:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:443:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:443:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', true, false>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, fal[  429.952664] serial8250: too much work for irq4
[  467s] se>();
[  467s]                 ^
[  467s] ../Ppu.cpp:497:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<true>' requested here
[  467s]                         RenderMode7<true>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, true, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:437:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x01', false, true>' requested here
[  467s]                 RenderTilemapMode7<1, forMainScreen, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false, true, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1841:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false, true, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1829:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false, true>' requested here
[  430.111074] serial8250: too much work for irq4
[  467s] 
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] ../Ppu.cpp:848:17: error: constexpr variable cannot have non-literal type 'const (lambda at ../Ppu.cpp:848:24)'
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                        ^
[  467s] ../Ppu.cpp:1839:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false, false, true>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, applyMosaic, true>();
[  467s]                 ^
[  467s] ../Ppu.cpp:1831:3: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false, false>' requested here
[  467s]                 RenderTilemapMode7<layerIndex, forMainScreen, processHighPriority, false>();
[  467s]                 ^
[  467s] ../Ppu.cpp:440:2: note: in instantiation of function template specialization 'Ppu::RenderTilemapMode7<'\x00', false, false>' requested here
[  467s]         RenderTilemapMode7<0, forMainScreen, false>();
[  467s]         ^
[  467s] ../Ppu.cpp:498:4: note: in instantiation of function template specialization 'Ppu::RenderMode7<false>' requested here
[  467s]                         RenderMode7<false>();
[  467s]                         ^
[  467s] ../Ppu.cpp:848:24: note: lambda closure types are non-literal types before C++17
[  467s]         constexpr auto clip = [](int32_t val) { return (val & 0x2000) ? (val | ~0x3ff) : (val & 0x3ff); };
[  467s]                               ^
[  467s] fatal error: too many errors emitted, stopping now [-ferror-limit=]
[  467s] 20 errors generated.
[  467s] make: *** [makefile:131: Core/obj.x64/Ppu.o] Error 1
```

https://build.opensuse.org/package/live_build_log/home:KAMiKAZOW:Test/mesen-s/openSUSE_Tumbleweed/x86_64

**Comment by SourMesen** [comment]
The best answer I have at the moment would be that somewhere between clang 6.0 (the version I use) and 7.7 (the version your log lists), restrictions were added that prevent this from working unless that C++17 flag is turned on, so your best bet would be to try changing the compilation flag in the makefile.

**Comment by SourMesen** [comment]
This should be fixed by b1a50434c97c574004af0e976ecb3533c6dc60a3
